common/src/main/java/dev/ftb/mods/ftblibrary/ui/misc/AbstractGroupedButtonListScreen.java

```diff
@@ -0,0 +1,226 @@
+package dev.ftb.mods.ftblibrary.ui.misc;
+
+import com.mojang.blaze3d.platform.InputConstants;
+import dev.ftb.mods.ftblibrary.icon.Color4I;
+import dev.ftb.mods.ftblibrary.icon.Icons;
+import dev.ftb.mods.ftblibrary.ui.Button;
+import dev.ftb.mods.ftblibrary.ui.Panel;
+import dev.ftb.mods.ftblibrary.ui.SimpleButton;
+import dev.ftb.mods.ftblibrary.ui.TextField;
+import dev.ftb.mods.ftblibrary.ui.Theme;
+import dev.ftb.mods.ftblibrary.ui.input.Key;
+import dev.ftb.mods.ftblibrary.ui.input.MouseButton;
+import dev.ftb.mods.ftblibrary.ui.misc.AbstractButtonListScreen;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.network.chat.Component;
+import org.lwjgl.glfw.GLFW;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static dev.ftb.mods.ftblibrary.util.TextComponentUtils.hotkeyTooltip;
+
+public abstract class AbstractGroupedButtonListScreen<G, E> extends AbstractButtonListScreen {
+    private final List<GroupData<G, E>> groups;
+    private final Map<G, Boolean> collapsed = new HashMap<>();
```

Comment:
Is this only used as a true/false test?  (i.e. not a tristate of true/false/absent ?)

If so, a `Set<G>` is preferable (in the set = true, not in the set = false)